### PR TITLE
datatrove: init at 0.2.0

### LINKED ISSUE #311862: Package request: python3xPackages.datatrove
**Project description**

From the repo's readme:

> DataTrove is a library to process, filter and deduplicate text data at a very large scale. It provides a set of prebuilt commonly used processing blocks with a framework to easily add custom functionality.

**Metadata**

* homepage URL: https://github.com/huggingface/datatrove/tree/main
* source URL: https://github.com/huggingface/datatrove/tree/main
* license: apache 2.0
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #311862

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/da/datatrove/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+}:
+python3Packages.buildPythonPackage rec {
+  pname = "datatrove";
+  version = "0.2.0";
```
Comment (isabelroses) [inline pkgs/by-name/da/datatrove/package.nix:8]: https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec
```suggestion
let
  version = "0.2.0";
in
python3Packages.buildPythonPackage {
  pname = "datatrove";
  inherit version;
```

**Review by isabelroses — COMMENTED** [review]
Other then that one nitpick seems good.

**Comment by isabelroses** [comment]
You should also squash you commits into one commit called `datatrove: init at 0.2.0`.

**Comment by luftmensch-luftmensch** [comment]
Shouldn't be this possible automatically?

**Review by isabelroses — APPROVED** [review]

**Comment by isabelroses** [comment]
> Shouldn't be this possible automatically?

Perhaps, I've always just squashed and force pushed. This probably depends on the reviews perspective.

**Review by Gerg-L — APPROVED** [review]

File: pkgs/by-name/da/datatrove/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  pythonImportsCheck = [ "datatrove" ];
+  meta = with lib; {
+    description = "Set of platform-agnostic customizable pipeline processing blocks for data processing";
+    homepage = "https://github.com/huggingface/datatrove";
+    changelog = "https://github.com/huggingface/datatrove/releases/tag/v${version}";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ luftmensch-luftmensch ];
+    platforms = platforms.all;
```
Comment (Gerg-L) [inline pkgs/by-name/da/datatrove/package.nix:40]:
```suggestion
  meta = {
    description = "Set of platform-agnostic customizable pipeline processing blocks for data processing";
    homepage = "https://github.com/huggingface/datatrove";
    changelog = "https://github.com/huggingface/datatrove/releases/tag/v${version}";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ luftmensch-luftmensch ];
    platforms = lib.platforms.all;
```

Comment (Gerg-L) [reply pkgs/by-name/da/datatrove/package.nix:40]: my only nitpick

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 311867` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>datatrove</li>
    <li>datatrove.dist</li>
  </ul>
</details>

File: pkgs/by-name/da/datatrove/package.nix (hunk 2 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/da/datatrove/package.nix]: Applied requested suggestion!

**Review by Gerg-L — APPROVED** [review]
LGTM

File: pkgs/by-name/da/datatrove/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+python3Packages.buildPythonPackage {
+  pname = "datatrove";
+  inherit version;
+  pyproject = true;
+  src = fetchFromGitHub {
+    owner = "huggingface";
+    repo = "datatrove";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-2NJja2yWeHOgo1pCuwHN6SgYnsimuZdK0jE8ucTH4r8=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/da/datatrove/package.nix:18]:
```suggestion
  pyproject = true;

  src = fetchFromGitHub {
    owner = "huggingface";
    repo = "datatrove";
    rev = "refs/tags/v${version}";
    hash = "sha256-2NJja2yWeHOgo1pCuwHN6SgYnsimuZdK0jE8ucTH4r8=";
  };

```

**Review by SuperSandro2000 — COMMENTED** [review]
Please run the tests with pytestCheckHook https://github.com/huggingface/datatrove/tree/main/tests

**Pull request merged**